jdaviz/configs/specviz2d/plugins/cross_dispersion_profile/cross_dispersion_profile.py

```diff
@@ -0,0 +1,313 @@
+        y_pixel, width, and the slider limits for selecting row/column where
+        the profile will be measured.
+        """
+        # self.dataset might not exist when app is setting itself up.
+        if hasattr(self, "dataset") and self.dataset.selected_obj is not None:
+            data = self.dataset.selected_obj
+            # default value for 'y_pixel' is middle of cross dispersion axis
+            self.y_pixel = math.floor(data.shape[0] / 2)
+            # default value for 'pixel' is middle of spectral axis
+            self.pixel = math.floor(data.shape[1] / 2)
+            self._pixel_to_wav()  # also set corresponding wavelength
+            # slider limits
+            self.max_y_pix = data.shape[0]
+            self.max_pix = data.shape[1]
+            # default use_full_width=True
+            self.use_full_width = True
+            # set appropriate default 'width' if use_full_width=False
+            self.width = data.shape[0]
+
+    @observe('pixel', 'sa_display_unit')
+    def _pixel_to_wav(self, event={}):
+        """
+        Calculate the corresponding wavelength for ``pixel``, if wcs is present,
+        when ``pixel`` is changed.
+        """
+        data = self.dataset.selected_obj
+        if data is not None:
+            if hasattr(data, 'wcs') and self.sa_display_unit != '':
+                wcs = self.dataset.selected_obj.wcs
+                wav = wcs.pixel_to_world(self.pixel)
+                self.wav = wav.to(u.Unit(self.sa_display_unit), u.spectral()).value
+
+    def _on_display_units_changed(self, event={}):
+        """
+        On flux display unit change from Unit Conversion plugin, re-compute
+        profile in new unit and update plot.
+
+        Note: re-measure profile in native data units rather than converting
+        currently computed profile so repeated conversions don't accumulate
+        precision errors.
+        """
+        if event.axis == 'flux':
+            if self.flux_display_unit == event.unit:
+                return
+            self.flux_display_unit = event.unit.to_string()
+
+        if event.axis == 'spectral':
+            if self.sa_display_unit == event.unit:
+                return
+            self.sa_display_unit = event.unit.to_string()
+
+            # convert ``pixel`` to new wavelength unit
+            self._pixel_to_wav()
+
+        # re-compute profile and update plot to new units
+        self.measure_cross_dispersion_profile(update_plot=True)
+
+    @property
+    def profile(self):
+        return self._profile
+
+    @property
+    def marks(self):
+        """
+        Access the marks created by this plugin in the spectrum-2d-viewer.
+        """
+        if self._marks:
+            return self._marks
+
+        if not self._tray_instance:
+            return {}
+
+        v2d = self.app.get_viewer_by_id('specviz2d-0')
+        v1d = self.app.get_viewer_by_id('specviz2d-1')

```

Comment:
you can use `self.spectrum_1d_viewers[0]` and `self.spectrum_2d_viewers[0]` for now and eventually we'll want to loop over the lists and handle this with mark collections so they're automatically added to new viewers, etc.